### PR TITLE
fix: 🐛 Fix resetForm not re-validating

### PR DESCRIPTION
**Pull request by aharpt**
Fix resetForm() not re-validating.  Fix suggested towards the bottom of the linked Github issue. https://github.com/jaredpalmer/formik/issues/3512.

https://www.loom.com/share/d302bc139c284c9e89e0764da6fbf4ba

Updated Loom: https://www.loom.com/share/a5ba4a97030748a0962cadfd409b1cc5

✅ Closes: #762

### CODE REVIEW

File: src/components/SQForm/SQFormButton.tsx
Comment (20BBrown14) [inline src/components/SQForm/SQFormButton.tsx:37]: Hm. This is def weird. Let's add a comment here with the relevant information about why we're doing this so we don't forget. A link to an issue  would be great so that we can see when/if it's fixed.

File: src/components/SQForm/SQFormButton.tsx
Comment (laurelbean) [inline src/components/SQForm/SQFormButton.tsx:6]: I think we need to use something like `unknown` for this default value. `void` is for functions, but you're using Values as a type for an object.

Comment (aharpt) [reply src/components/SQForm/SQFormButton.tsx:6]: @laurelbean Sounds good.  I'll look into updating this.

**Comment by laurelbean** [comment]
`initialValues` is a value that you can get from `useFormikContext` which can be gotten from `useFormButton`.
So this can be simplified into something more like:
```
const {. . ., initialValues, setValues} = useFormButton({. . .});

const formReset = (event: React.MouseEvent<HTMLButtonElement>) => {
    handleReset(event);  //pass through event
    setValues(initialValues); //not sure if this should go 1st or 2nd
  };

  const getClickHandler = (event: React.MouseEvent<HTMLButtonElement>) => {
    if (isResetButton) {
      return formReset;
    } 
    . . .
  };
  ```
  
  I haven't tested it, so might need some adjustments.
  
  This doesn't solve the helper text issue though. You might want to look into trying adding in `validateForm` or `setErrors` to get the form in the right state for the helper text.

**Comment by laurelbean** [comment]
Another option we could potentially pursue would be downgrading the Formik version to one that doesn't have this reset issue. As long as it doesn't re-introduce other big issues.

**Comment by aharpt** [comment]
@laurelbean @20BBrown14 I am hoping that I have finally gotten to the bottom of the issue and found a decent solution.  Here is an updated loom: https://www.loom.com/share/a5ba4a97030748a0962cadfd409b1cc5

cc: @lucas-homer @hvilander

**Review by hvilander — APPROVED** [review]

**Comment by hvilander** [comment]
Great work Aaron!!

**Review by laurelbean — APPROVED** [review]

**Review by 20BBrown14 — APPROVED** [review]

**Review by lucas-homer — APPROVED** [review]

**Review by jamesgriffith9519 — APPROVED** [review]
This looks similar to how we addressed the issue locally in the AQE2 project and from what I can tell it seems to work well!

**Comment by laurelbean** [comment]
Unless someone needs this soon, I think this could wait until after our Mui 5 update to try and avoid merge conflicts. If it's needed sooner though I have no issues pushing it.

**Comment by hvilander** [comment]
> Unless someone needs this soon, I think this could wait until after our Mui 5 update to try and avoid merge conflicts. If it's needed sooner though I have no issues pushing it.

Agreed, at least I don't think this is immediately urgent. Where I bumped into it is long from being put into production.

**Pull request merged**
